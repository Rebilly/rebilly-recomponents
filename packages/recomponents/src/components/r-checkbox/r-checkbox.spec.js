import {mount, shallowMount} from '@vue/test-utils';
import {renderToString} from '@vue/server-test-utils';
import RCheckbox from './r-checkbox.vue';

const $t = () => 'custom label';

describe('r-checkbox.vue', () => {
    it('renders props.label when passed', () => {
        const wrapper = shallowMount(RCheckbox, {
            propsData: {label: 'custom label'},
            mocks: {$t},
        });

        expect(wrapper.text()).toMatch('custom label');
    });

    it('should render via SSR and match snapshot', async () => {
        const wrapper = renderToString(RCheckbox, {
            propsData: {
                name: 'snapshot',
                label: 'Custom label',
                id: 'id1234',
            },
            mocks: {$t},
        });

        expect(wrapper).toMatchSnapshot();
    });

    it('should render component with fuzzy prop', () => {
        const wrapper = mount(RCheckbox, {
            propsData: {
                fuzzy: true,
            },
            mocks: {$t},
        });

        expect(wrapper.find('use').attributes().href).toBe('#icon-dash-l');
    });

    it('should render component with disabled prop', () => {
        const wrapper = mount(RCheckbox, {
            propsData: {
                disabled: true,
            },
            mocks: {$t},
        });
        expect(wrapper.find('input').attributes().disabled).not.toBe(undefined);
    });

    it('should handle change event', () => {
        const wrapper = mount(RCheckbox, {
            propsData: {
                name: 'checkbox',
                label: 'Label',
                selected: [false, true],
            },
            mocks: {$t},
        });

        wrapper.find('.r-field-checkbox').trigger('change');
        expect(wrapper.emitted().input).toBeTruthy();
    });
});
