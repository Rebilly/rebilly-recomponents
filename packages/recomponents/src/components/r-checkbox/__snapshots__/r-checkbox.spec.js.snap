// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`r-checkbox.vue should render via SSR and match snapshot 1`] = `
<div role="checkbox" name="snapshot" class="r-checkbox"><label for="id1234" class="r-field-label r-field-label-toggle">Custom label</label> <input type="checkbox" id="id1234" value="on" class="r-field-checkbox"> <span class="r-field-checkbox-style"><svg class="r-icon r-icon-20"><use xlink:href="#icon-checkbox-checkmark" xmlns:xlink="http://www.w3.org/1999/xlink"></use></svg></span>
  <!---->
</div>
`;
